NGitLab.Mock/Clients/ProtectedBranchClient.cs

```diff
@@ -13,24 +14,41 @@ public ProtectedBranchClient(ClientContext context, int projectId)
             _projectId = projectId;
         }
 
-        public ProtectedBranch GetProtectedBranch(string branchName)
+        public Models.ProtectedBranch GetProtectedBranch(string branchName)
         {
-            throw new NotImplementedException();
+            using (Context.BeginOperationScope())
+            {
+                var project = GetProject(_projectId, ProjectPermission.View);
+                return project.ProtectedBranches.First(b => b.Name.Equals(branchName, StringComparison.Ordinal)).ToProtectedBranchClient();
+            }
         }
 
-        public ProtectedBranch[] GetProtectedBranches(string search = null)
+        public Models.ProtectedBranch[] GetProtectedBranches(string search)
         {
-            throw new NotImplementedException();
+            using (Context.BeginOperationScope())
+            {
+                var project = GetProject(_projectId, ProjectPermission.View);
+                return project.ProtectedBranches.Select(b => b.ToProtectedBranchClient()).ToArray();
+            }
         }
 
-        public ProtectedBranch ProtectBranch(BranchProtect branchProtect)
+        public Models.ProtectedBranch ProtectBranch(BranchProtect branchProtect)
         {
             throw new NotImplementedException();
```

Comment:
I think you can implement the method using something like:

````
var protectedBranch = project.ProtectedBranches.FirstOrDefault(b => b.Name == branchName);
if(protectedBranch == null)
{
    protectedBranch = new();
    project.ProtectedBranches.Add(protectedBranch);
}

// TODO Use actual properties
protectedBranch.Prop = branchProtect.Prop;
protectedBranch.Prop = branchProtect.Prop;
````

Reply:
Desole. I misunderstood the previous comment. Corrected :)